### PR TITLE
Some gui improvements

### PR DESCRIPTION
**Pull request by azarovh**
- Copy transaction id
- Added validation to all input fields
- Display pool ids for delegations
Screenshot 2024-01-26 at 12 17 47

- Display margin ration and cost per block for pools
Screenshot 2024-01-26 at 12 17 39

### CODE REVIEW

File: common/src/primitives/per_thousand.rs [hunk 1]
```diff
@@ -57,6 +57,13 @@ impl PerThousand {
         let result = Self::new(value)?;
         Some(result)
     }
+
+    pub fn into_decimal_str(&self) -> String {
+        format!(
+            "{}%",
+            Amount::from_atoms(self.0.into()).into_fixedpoint_str(1)
+        )
+    }
```
Comment (iljakuklic) [inline common/src/primitives/per_thousand.rs:66]: Would it make sense to put this into a `Display` trait impl?

File: node-gui/src/main_window/main_widget/tabs/wallet/addresses.rs [hunk 2]
```diff
@@ -23,6 +23,8 @@ use crate::backend::messages::AccountInfo;
 
 use super::WalletMessage;
 
+const NEW_ADDRESS_TOOLTIP_TEXT: &str = "TODO";
```
Comment (iljakuklic) [inline node-gui/src/main_window/main_widget/tabs/wallet/addresses.rs:26]: Can we put something else here? Even if it's just something like "Generate a new address".

File: node-gui/src/main_window/main_widget/tabs/wallet/delegation.rs [hunk 3]
```diff
@@ -34,12 +35,15 @@ const POOL_ID_TOOLTIP_TEXT: &str =
...
 
-const SEND_DELEGATION_ADDRESS_TOOLTIP_TEXT: &str = " The address that will be receiving the coins";
+const SEND_DELEGATION_ADDRESS_TOOLTIP_TEXT: &str = "The address that will be receiving the coins";
 const SEND_DELEGATION_AMOUNT_TOOLTIP_TEXT: &str =
     "The amount that will be taken away from the delegation";
 const SEND_DELEGATION_ID_TOOLTIP_TEXT: &str =
     "The delegation id, from which the delegated coins will be taken";
 
+const SEND_TO_ADDRESS_BUTTON_TOOLTIP_TEXT: &str = "TODO";
+const CREATE_DELEGATION_BUTTON_TOOLTIP_TEXT: &str = "TODO";
```
Comment (iljakuklic) [inline node-gui/src/main_window/main_widget/tabs/wallet/delegation.rs:45]: More TODOs...

File: node-gui/src/main_window/main_widget/tabs/wallet/mod.rs [hunk 4]
```diff
@@ -360,6 +361,7 @@ impl WalletTab {
                 self.account_state.send_delegation_id.clear();
                 Command::none()
             }
+            WalletMessage::NoOp => Command::none(),
```
Comment (iljakuklic) [inline node-gui/src/main_window/main_widget/tabs/wallet/mod.rs:364]: Why is this needed? What's the difference between sending a NoOp and not sending anything?

**Review by iljakuklic — APPROVED** [review]
Looks fine to me but I'm not very familiar with gui so don't take this too seriously. A couple of comments / questions below.

File: node-gui/src/main_window/main_widget/tabs/wallet/delegation.rs (hunk 3 above)
Comment (TheQuantumPhysicist) [reply node-gui/src/main_window/main_widget/tabs/wallet/delegation.rs:45]: I'm working on it as we speak.

File: node-gui/src/main_window/main_widget/tabs/wallet/addresses.rs (hunk 2 above)
Comment (TheQuantumPhysicist) [reply node-gui/src/main_window/main_widget/tabs/wallet/addresses.rs:26]: Please come back in 15 minutes, you'll see all the texts. I do need scrutiny once I'm done.

File: node-gui/src/main_window/main_widget/tabs/wallet/mod.rs (hunk 4 above)
Comment (azarovh) [reply node-gui/src/main_window/main_widget/tabs/wallet/mod.rs:364]: There are cases where I have to send a message, like in closures with input validation: Fn(String) -> Message. So NoOp is a workaround there

File: common/src/primitives/per_thousand.rs (hunk 1 above)
Comment (TheQuantumPhysicist) [reply common/src/primitives/per_thousand.rs:66]: It could be both a percentage and decimal. I wouldn't use Display.

File: node-gui/src/main_window/main_widget/tabs/wallet/mod.rs (hunk 4 above)
Comment (iljakuklic) [reply node-gui/src/main_window/main_widget/tabs/wallet/mod.rs:364]: Ok. I don't have a strong opinion about this.

**Review by TheQuantumPhysicist — APPROVED** [review]
